packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts

```diff
@@ -0,0 +1,58 @@
+import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {logProductRecommendationOpen} from '../../../features/product-listing/product-listing-analytics';
+import {pushRecentResult} from '../../../features/product-listing/product-listing-recent-results';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+  InteractiveResultCoreOptions,
+  InteractiveResultCoreProps,
+} from '../../core/interactive-result/headless-core-interactive-result';
+
+export interface InteractiveResultOptions extends InteractiveResultCoreOptions {
+  /**
+   * The query result.
+   */
+  result: ProductRecommendation;
+}
+
+export interface InteractiveResultProps extends InteractiveResultCoreProps {
+  /**
+   * The options for the `InteractiveResult` controller.
+   * */
+  options: InteractiveResultOptions;
+}
+
+/**
+ * The `InteractiveResult` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a query result.
```

Comment:
```suggestion
 * The `InteractiveResult` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a product.
```

Reply:
This makes me wonder what other desirable side effects this controller could ever trigger...

Reply:
I just discussed it with @olamothe and I now see that it does other stuff under the hood such as handling long presses on mobile, or multiple clicks

Here's an updated suggestion for the documentation:

```
The `InteractiveProduct` controller provides an interface for handling long presses, multiple clicks, etc. to ensure that Coveo usage analytics events are logged properly when a user selects a product.
```